pkg/pgmodel/sql_bench_test.go

```diff
@@ -0,0 +1,262 @@
+package pgmodel
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+const (
+	labelCount      = 5
+	metricName      = "benchmark_metric"
+	otherMetricName = "other_benchmark_metric"
+)
+
+func BenchmarkGetSeriesIDForKeyValueArrayExistingSeries(b *testing.B) {
+	b.StopTimer()
+	withDB(b, "bench_1", func(db *pgxpool.Pool, t testing.TB) {
+		err := createMetricTableName(db, metricName)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		keys, values := generateKeysAndValues(labelCount, "label")
+		err = getSeriesIDForKeyValueArray(db, metricName, keys, values)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var bench *testing.B
+		var ok bool
+		if bench, ok = t.(*testing.B); !ok {
+			t.Fatal("Not a benchmarking instance, stopping benchmark")
+		}
+
+		bench.ResetTimer()
+		bench.StartTimer()
+		for n := 0; n < b.N; n++ {
+			err = getSeriesIDForKeyValueArray(db, metricName, keys, values)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+		bench.StopTimer()
+	})
+}
+
+func BenchmarkGetSeriesIDForKeyValueArrayNewSeriesExistingLabels(b *testing.B) {
+	b.StopTimer()
+	withDB(b, "bench_2", func(db *pgxpool.Pool, t testing.TB) {
+		err := createMetricTableName(db, metricName)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		keys, values := generateKeysAndValues(b.N+labelCount, "label")
+		err = getSeriesIDForKeyValueArray(db, metricName, keys, values)
```

Comment:
This calls get_series_id_for_key_value_array() with really big keys and values slices. This function is meant to be called for keys and values for one series. also don't we want this going to otherMetricName. So I think this should be 
```
		for n := 0; n < b.N; n++ {
			err = getSeriesIDForKeyValueArray(db, otherMetricName, keys[n:n+labelCount], values[n:n+labelCount])
			if err != nil {
				t.Fatal(err)
			}
		}
```

Reply:
This part isn't in the benchmark timing. I just do it like so to insert all the labels at once. Theoretically, doing it this way or your way, shouldn't matter for the sql performance?